sentry-actix/src/lib.rs

```diff
@@ -191,21 +196,63 @@ where
 
     fn new_transform(&self, service: S) -> Self::Future {
         ok(SentryMiddleware {
-            service,
+            service: Rc::new(service),
             inner: self.clone(),
         })
     }
 }
 
 /// The middleware for individual services.
 pub struct SentryMiddleware<S> {
-    service: S,
+    service: Rc<S>,
     inner: Sentry,
 }
 
+async fn should_capture_request_body(headers: &HeaderMap) -> bool {
+    headers
+        .get("content-type")
+        .and_then(|h| h.to_str().ok())
+        .map(|s| s.to_lowercase())
+        .is_some_and(|ct| {
+            ct.contains("application/json") || ct.contains("application/x-www-form-urlencoded")
+        })
+}
+
+/// Extract a body from the HTTP request
+async fn body_from_http(req: &mut ServiceRequest) -> Result<BytesMut, PayloadError> {
+    let mut stream = req.take_payload();
+
+    let mut body = BytesMut::new();
+    while let Some(chunk) = futures_util::StreamExt::next(&mut stream).await {
+        let chunk = chunk?;
+        body.extend_from_slice(&chunk);
+    }
+    let (_, mut orig_payload) = actix_http::h1::Payload::create(true);
+    orig_payload.unread_data(body.clone().freeze());
+    req.set_payload(actix_web::dev::Payload::from(orig_payload));
+
+    Ok::<_, PayloadError>(body)
+}
+
+async fn capture_request_body(
+    req: &mut ServiceRequest,
+    max_size: MaxRequestBodySize,
+) -> Option<String> {
+    let request_body = body_from_http(req).await.unwrap();
+    let request_body_str = String::from_utf8_lossy(&request_body).into_owned();
+
```

Comment:
you should do the size check before you are doing the utf-8 conversion. Otherwise this is just wasted work in case it fails the size-check.

Reply:
It's Done by the new implementation of `should_capture_request_body` method now checks the `Content-Length`, `Content-Type`, and ensures that `transfer_encoding` does not include `chunked`. However, with this update, I no longer retain the empty string when the body is too large, meaning it won’t be displayed in the Sentry issue. Should I modify this to keep the empty string in those cases?